tests/helpers/common/Chocolatey/Invoke-Choco.ps1

```diff
@@ -33,28 +33,43 @@
 
             $remainingArguments
         )
-
     }
     end {
+        $stopwatch = [System.Diagnostics.Stopwatch]::new()
+        $stopwatch.Start()
+
         $output = if ($PipelineInput) {
             $PipelineInput | & $chocoPath @arguments
         }
         else {
             & $chocoPath @arguments
         }
 
+        $stopwatch.Stop()
+        # We do not use env:ChocolateyInstall here as it is not guaranteed due to snapshots.
+        # Saving into the default Chocolatey install location as .log so Test Kitchen can pick up the file.
+        [PSCustomObject]@{
+            Date       = Get-Date -Format 'o'
+            Duration   = $stopwatch.Elapsed
+            Invocation = ($MyInvocation.PositionMessage -split "`r`n")[0]
+            Arguments  = ($Arguments -join ' ') -replace "`r`n", ''
+            ExitCode   = $LastExitCode
+        } | Export-Csv -Path $env:ALLUSERSPROFILE/chocolatey/logs/testInvocations.log -NoTypeInformation -Append
+
         [PSCustomObject]@{
             # We trim all the lines, so we do not take into account
             # trimming the lines when asserting, and that extra whitespace
             # is not considered in our assertions.
-            Lines    = if ($output) {
+            Lines     = if ($output) {
                 $output.Trim()
             }
             else {
                 @()
             }
-            String   = $output -join "`r`n"
-            ExitCode = $LastExitCode
+            String    = $output -join "`r`n"
+            ExitCode  = $LastExitCode
+            Arguments = $arguments

```

Comment:
As we should already know the arguments used, let us not pass the arguments back to the test being run.

Reply:
The scenario where the arguments being returned would be beneficial are on things like the Intune setup where we do a series of `Invoke-Choco` that we store in an array, then if any of them exit with failure we throw an exception and provide the output, but we don't know which ones failed. I could be convinced to change this and throw at a place where we know these details...

Reply:
Since we don't currently have a use case for this specific scenario, I've taken it out. It's also probably better for us to update the calling code to know what might have failed in a different way.